File: MATLAB/Gaussian Filter.m
img = imread('/MATLAB Drive/noisy_image.jpg');
filterSize = 10;
sigma = 15;
h = fspecial('gaussian', filterSize, sigma);
[rows, cols, numChannels] = size(img);
filteredImg = zeros(rows, cols, numChannels, 'like', img);
for channel = 1:numChannels
    filteredImg(:, :, channel) = imfilter(img(:, :, channel), h, 'replicate');
end
figure('Position', [100, 100, 600, 800]);
subplot(2, 1, 1);
imshow(img);
title('Original Image');
subplot(2, 1, 2);
imshow(filteredImg);
title('Filtered Image with Gaussian Filter');

File: Python/Gaussian Filter.py
import cv2
import numpy as np
from matplotlib import pyplot as plt

# Read the image
img = cv2.imread('Enter the path to the image here')  
img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)    

# Define filter parameters
filter_size = 11
sigma = 15

# Apply Gaussian filter
filtered_img = np.zeros_like(img)
for channel in range(img.shape[2]):
    filtered_img[:, :, channel] = cv2.GaussianBlur(img[:, :, channel], (filter_size, filter_size), sigma)

# Display original and filtered images
plt.figure(figsize=(6, 8))

plt.subplot(2, 1, 1)
plt.imshow(img)
plt.title('Original Image')
plt.axis('off')

plt.subplot(2, 1, 2)
plt.imshow(filtered_img)
plt.title('Filtered Image with Gaussian Filter')
plt.axis('off')

plt.tight_layout()
plt.show()
